Tests automatically failing in windows environment

On my local clone of sympy, without modifications, if I launch:

`pytest sympy/testing/tests/test_runtests_pytest.py --tb=short`

I get:

`sympy\testing\tests\test_runtests_pytest.py ..........FFFFFFFFFFFF                                               [100%]`

a major part of the tests in that module fail, and the failures look like this:

```
====================================================== FAILURES =======================================================
__________________ TestUpdateArgsWithPaths.test_multiple_paths_from_non_root[paths0-expected_paths0] __________________
sympy\testing\tests\test_runtests_pytest.py:104: in test_multiple_paths_from_non_root
    assert expected in arg
E   AssertionError: assert 'doc/src/modules/utilities' in 'C:\\Users\\andrea\\Documents\\GitHub\\sympy\\doc\\src\\modules\\utilities'
_______________________________ TestUpdateArgsWithPaths.test_string_as_keyword[paths0] ________________________________
sympy\testing\tests\test_runtests_pytest.py:125: in test_string_as_keyword
    assert expected in arg
E   AssertionError: assert 'sympy/physics/mechanics/tests/test_kane3.py::test_bicycle' in 'C:\\Users\\andrea\\Documents\\GitHub\\sympy\\sympy\\physics\\mechanics\\tests\\test_kane3.py::test_bicycle'
_______________________________ TestUpdateArgsWithPaths.test_string_as_keyword[paths1] ________________________________
sympy\testing\tests\test_runtests_pytest.py:125: in test_string_as_keyword
    assert expected in arg
E   AssertionError: assert 'sympy/physics/mechanics/tests/test_kane3.py::test_bicycle' in 'C:\\Users\\andrea\\Documents\\GitHub\\sympy\\sympy\\physics\\mechanics\\tests\\test_kane3.py::test_bicycle'
_______________________________ TestUpdateArgsWithPaths.test_string_as_keyword[paths2] ________________________________
sympy\testing\tests\test_runtests_pytest.py:125: in test_string_as_keyword
    assert expected in arg
E   AssertionError: assert 'sympy/physics/mechanics/tests/test_kane3.py::test_bicycle' in 'C:\\Users\\andrea\\Documents\\GitHub\\sympy\\sympy\\physics\\mechanics\\tests\\test_kane3.py::test_bicycle'
_______________________________ TestUpdateArgsWithPaths.test_string_as_keyword[paths3] ________________________________
sympy\testing\tests\test_runtests_pytest.py:125: in test_string_as_keyword
    assert expected in arg
E   AssertionError: assert 'sympy/physics/mechanics/tests/test_kane3.py::test_bicycle' in 'C:\\Users\\andrea\\Documents\\GitHub\\sympy\\sympy\\physics\\mechanics\\tests\\test_kane3.py::test_bicycle'
_______________________________ TestUpdateArgsWithPaths.test_string_as_keyword[paths4] ________________________________
sympy\testing\tests\test_runtests_pytest.py:125: in test_string_as_keyword
    assert expected in arg
E   AssertionError: assert 'sympy/physics/mechanics/tests/test_kane3.py::test_bicycle' in 'C:\\Users\\andrea\\Documents\\GitHub\\sympy\\sympy\\physics\\mechanics\\tests\\test_kane3.py::test_bicycle'
_______________________________ TestUpdateArgsWithPaths.test_integer_as_keyword[paths0] _______________________________
sympy\testing\tests\test_runtests_pytest.py:145: in test_integer_as_keyword
    assert expected in arg
E   AssertionError: assert 'sympy/core/tests/test_sympify.py::test_issue_3538' in 'C:\\Users\\andrea\\Documents\\GitHub\\sympy\\sympy\\core\\tests\\test_sympify.py::test_issue_3538'
_______________________________ TestUpdateArgsWithPaths.test_integer_as_keyword[paths1] _______________________________
sympy\testing\tests\test_runtests_pytest.py:145: in test_integer_as_keyword
    assert expected in arg
E   AssertionError: assert 'sympy/core/tests/test_sympify.py::test_issue_3538' in 'C:\\Users\\andrea\\Documents\\GitHub\\sympy\\sympy\\core\\tests\\test_sympify.py::test_issue_3538'
_______________________________ TestUpdateArgsWithPaths.test_integer_as_keyword[paths2] _______________________________
sympy\testing\tests\test_runtests_pytest.py:145: in test_integer_as_keyword
    assert expected in arg
E   AssertionError: assert 'sympy/core/tests/test_sympify.py::test_issue_3538' in 'C:\\Users\\andrea\\Documents\\GitHub\\sympy\\sympy\\core\\tests\\test_sympify.py::test_issue_3538'
_______________________________ TestUpdateArgsWithPaths.test_integer_as_keyword[paths3] _______________________________
sympy\testing\tests\test_runtests_pytest.py:145: in test_integer_as_keyword
    assert expected in arg
E   AssertionError: assert 'sympy/core/tests/test_sympify.py::test_issue_3538' in 'C:\\Users\\andrea\\Documents\\GitHub\\sympy\\sympy\\core\\tests\\test_sympify.py::test_issue_3538'
___________________________________ TestUpdateArgsWithPaths.test_multiple_keywords ____________________________________
sympy\testing\tests\test_runtests_pytest.py:158: in test_multiple_keywords
    assert expected in arg
E   AssertionError: assert 'sympy/core/tests/test_sympify.py::test_issue_3538' in 'C:\\Users\\andrea\\Documents\\GitHub\\sympy\\sympy\\core\\tests\\test_sympify.py::test_issue_3538'
____________________________ TestUpdateArgsWithPaths.test_keyword_match_in_multiple_files _____________________________
sympy\testing\tests\test_runtests_pytest.py:171: in test_keyword_match_in_multiple_files
    assert expected in arg
E   AssertionError: assert 'sympy/integrals/tests/test_heurisch.py::test_heurisch_symbolic_coeffs_1130' in 'C:\\Users\\andrea\\Documents\\GitHub\\sympy\\sympy\\integrals\\tests\\test_heurisch.py::test_heurisch_symbolic_coeffs_1130'

```

So basically the tests fail because on windows you have this **\\** instead of  this **/** in the path.

--- a/sympy/testing/tests/test_runtests_pytest.py
+++ b/sympy/testing/tests/test_runtests_pytest.py
@@ -1,6 +1,6 @@
 import pathlib
 from typing import List
-
+from pathlib import Path
 import pytest
 
 from sympy.testing.runtests_pytest import (
@@ -101,7 +101,7 @@ class TestUpdateArgsWithPaths:
         args = update_args_with_paths(paths=paths, keywords=None, args=[])
         assert len(args) == len(expected_paths)
         for arg, expected in zip(sorted(args), expected_paths):
-            assert expected in arg
+            assert Path(expected).as_posix() in Path(arg).as_posix()
 
     @staticmethod
     @pytest.mark.parametrize(
@@ -122,7 +122,7 @@ class TestUpdateArgsWithPaths:
         expected_args = ['sympy/physics/mechanics/tests/test_kane3.py::test_bicycle']
         assert len(args) == len(expected_args)
         for arg, expected in zip(sorted(args), expected_args):
-            assert expected in arg
+            assert Path(expected).as_posix() in Path(arg).as_posix()
 
     @staticmethod
     @pytest.mark.parametrize(
@@ -142,7 +142,7 @@ class TestUpdateArgsWithPaths:
         expected_args = ['sympy/core/tests/test_sympify.py::test_issue_3538']
         assert len(args) == len(expected_args)
         for arg, expected in zip(sorted(args), expected_args):
-            assert expected in arg
+            assert Path(expected).as_posix() in Path(arg).as_posix()
 
     @staticmethod
     def test_multiple_keywords():
@@ -155,7 +155,7 @@ class TestUpdateArgsWithPaths:
         ]
         assert len(args) == len(expected_args)
         for arg, expected in zip(sorted(args), expected_args):
-            assert expected in arg
+            assert Path(expected).as_posix() in Path(arg).as_posix()
 
     @staticmethod
     def test_keyword_match_in_multiple_files():
@@ -168,4 +168,4 @@ class TestUpdateArgsWithPaths:
         ]
         assert len(args) == len(expected_args)
         for arg, expected in zip(sorted(args), expected_args):
-            assert expected in arg
+            assert Path(expected).as_posix() in Path(arg).as_posix()

